Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -592,10 +592,21 @@ protected async Task HandleReadData( CancellationToken cancellationToken )
 
         protected async ValueTask<ItemsProviderResult<TItem>> HandleVirtualizeReadData( ItemsProviderRequest request )
         {
-            CurrentPage = request.StartIndex + 1;
+            var itemIndex = request.StartIndex;
+            var requestItems = Math.Min( request.Count, TotalItems.Value - itemIndex );
+
+            PageSize = requestItems;
+            //TODO: Do we need to introduce a new API that takes in a startIndex?

```

Comment:
I'm not sure what this does or why we need a new API. Can you explain the background?

Reply:
Virtualize API has a delegate that tells you two parameters.
I want the data from this index forward and I want this quantity of data. 
You answer with the data and total data. So ReadData already does this and we can take advantage of it, by passing in the Virtualize request to read data, However instead of current page it asks for current index, and that is why we need a new parameter. We don't exactly need to expose a new api, that would be only for separation of concerns... Which again I really don't think we need to separate, was asking your opinion. 

So you can basically use pagesize for the quantity of data.
But have to introduce a new start index for the ReadData arguments so that the developer can use the start index instead of page size.

There might be left over code from testing with trying to make it work with page, but I don't think it will ever be possible to do it 100% from what I've tried.

Reply:
OK, I understand now. Yeah, the new parameter makes sense. Otherwise users would have to to calculate index each time and this way we can help them a little.

Reply:
And I was thinking... Since we already introduce start idx might be more clear to just introduce VirtualizeRequestCount instead of reusing page size, since we couldn't straight up get it working with the pagination arguments. 
Meaning VirtualizeRequestStartIndex and VirtualizeRequestCount are the two new ReadData arguments being proposed.and maybe an argument flag that tells if the ReadData call is a manual one or a Virtualize one which should help people building grids that support both options.

Reply:
Makes sense. Just call them `VirtualizeStartIndex` and `VirtualizeCount` instead.

Maybe have an `enum ReadDataMode { Virtualize, Manual }`? Or if `VirtualizeStartIndex` and `VirtualizeCount` are not-null we can assume virtualize mode is on.

Reply:
I think I like the enum more then the nullable which can hide bugs, and the developer doesn't have to check for nullabikity which isn't clear and as readable / clean as using the enum. 
The enum makes it clear that the Datagrid wanted either a Virtualize or Manual request. Seems good to me.

Reply:
Also do the users outside of the data grid's internal api naming knowledge know that manual is the same as a paged request? Maybe the manual enum could be paging or page... Not sure.

Reply:
Yeah makes more sense to have enum. And naming it Paging instead of Manual is much better